packages/persoo/src/Component/Persoo/Client.php

```diff
@@ -0,0 +1,233 @@
+            '&offerID=' . $this->getOfferId($action) .
+            '&locationID=' . $this->getLocationId($action) .
+            '&query=' . urlencode('"' . $query . '"') .
+            '&itemsPerPage=' . $limit .
+            '&page=' . $offset .
+            '&index=' . urlencode($index) .
+            '&_vid=' . urlencode($browserId) .
+            ($filter !== [] ? '&boolQuery=' . urlencode(json_encode($filter, JSON_THROW_ON_ERROR)) : '') .
+            '&idsOnly=' . $this->getOnlyIds();
+    }
+
+    /**
+     * @return string
+     */
+    protected function getEnvironmentId(): string
+    {
+        return $this->persooIsProductionMode === true ? 'p' : 'test';
+    }
+
+    /**
+     * @return int
+     */
+    protected function getTimestampInMilliseconds(): int
+    {
+        return (int)floor(microtime(true) * 1000);
+    }
+
+    /**
+     * @return string
+     */
+    protected function getWebserverName(): string
+    {
+        return $this->requestStack->getMainRequest()->server->get('SERVER_NAME');
+    }
+
+    /**
+     * @param string $action
+     * @param string $name
+     * @return string
+     */
+    protected function getParameterByActionAndName(string $action, string $name): string
+    {
+        $parameterName = static::PERSOO_PARAMETER_PREFIX . '.' . $action . '.' . $name;
+
+        if (!$this->parameterBag->has($parameterName)) {
+            throw new PersooIndexNotRecognizedException(
+                sprintf('Persoo ENV variable for event "%s" with name "%s" or parameter "%s" is not set.', $action, $name, $parameterName),
+            );
+        }
+
+        return $this->parameterBag->get($parameterName);
+    }
+
+    /**
+     * @param string $action
+     * @return string
+     */
+    protected function getAlgorithmId(string $action): string
+    {
+        return $this->getParameterByActionAndName($action, static::PERSOO_PARAMETER_ALGORITHM_ID);
+    }
+
+    /**
+     * @param string $action
+     * @return string
+     */
+    protected function getOfferId(string $action): string
+    {
+        return $this->getParameterByActionAndName($action, static::PERSOO_PARAMETER_OFFER_ID);
+    }
+
+    /**
+     * @param string $action
+     * @return string
+     */
+    protected function getLocationId(string $action): string
+    {
+        return $this->getParameterByActionAndName($action, static::PERSOO_PARAMETER_LOCATION_ID);
+    }
+
+    /**
+     * @return string
+     */
+    protected function getOnlyIds(): string
+    {
+        return 'false';
+    }
+
+    /**
+     * @param string $index
+     * @return string
+     */
+    protected function getIdFieldNameByIndex(string $index): string
+    {
+        if ($index === 'products') {
+            return 'id';
+        }
+
+        if ($index === 'categories') {
+            return 'categoryId';
+        }
+
+        throw new PersooIndexNotRecognizedException(sprintf('Persoo index "%s" is not recognized.', $index));
```

Comment:
what about articles?

Reply:
I have no information on what name of the column should be used yet, so I could not implement it.